tests/test_basic.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Faculty Science Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hiplot
+import pandas as pd
+
+import hiplot_mlflow
+import pytest
+
+
+def test_wrong_schema():
+    """Test passing unsupported schema.
+    """
+    with pytest.raises(hiplot.ExperimentFetcherDoesntApply):
+        hiplot_mlflow.fetch_by_uri("something://")
+
+
+def test_unsupported_reference_type():
+    """Test passing correct schema and unsupported reference type.
+    """
+    with pytest.raises(hiplot.ExperimentFetcherDoesntApply):
+        hiplot_mlflow.fetch_by_uri(f"mlflow://something/else")
+
+
+def test_dataframe_to_experiment():
+    """Test the conversion from dataframe to HiPlot experiment,
+    with or without tags.
+    """
+    data = {
+        "run_id": [
+            "123e4567-e89b-12d3-a456-426655440000",
+            "123e4567-e89b-12d3-a456-426655440001",
+        ],
+        "params.first": [0, 1],
+        "metrics.second": [2, 4],
+        "tags.test": ["yes", "no"],
+    }
+    df = pd.DataFrame(
+        data, columns=["run_id", "params.first", "metrics.second", "tags.test"]
+    )
+
+    exp_notags = hiplot.Experiment()
+    dp1 = hiplot.Datapoint(
+        uid="123e4567-e89b-12d3-a456-426655440000", values={"first": 0, "second": 2}
+    )
+    exp_notags.datapoints.append(dp1)
+    dp2 = hiplot.Datapoint(
+        uid="123e4567-e89b-12d3-a456-426655440001", values={"first": 1, "second": 4}
+    )
+    exp_notags.datapoints.append(dp2)
+    test_notags = hiplot_mlflow._create_experiment_from_dataframe(
+        df, include_tags=False
+    )
+    for i, dp in enumerate(test_notags.datapoints):
+        assert dp.uid == exp_notags.datapoints[i].uid
+        assert dp.values == exp_notags.datapoints[i].values
+
+    exp_tags = hiplot.Experiment()
+    dp3 = hiplot.Datapoint(
+        uid="123e4567-e89b-12d3-a456-426655440000",
+        values={"first": 0, "second": 2, "test": "yes"},
+    )
+    exp_tags.datapoints.append(dp3)
+    dp4 = hiplot.Datapoint(
+        uid="123e4567-e89b-12d3-a456-426655440001",
+        values={"first": 1, "second": 4, "test": "no"},
+    )
+    exp_tags.datapoints.append(dp4)
+    test_tags = hiplot_mlflow._create_experiment_from_dataframe(df, include_tags=True)
+    for i, dp in enumerate(test_tags.datapoints):
+        assert dp.uid == exp_tags.datapoints[i].uid
+        assert dp.values == exp_tags.datapoints[i].values
```

Comment:
This is a good use case for `zip`:

```suggestion
    for test_dp, exp_dp in zip(test_tags.datapoints, exp_tags.datapoints):
        assert test_dp.uid == exp_dp.uid
        assert test_dp.values == exp_dp.values
```

Reply:
Ah, indeed, cheers!